Assets/Scripts/Framework/Gameplay/MiniGames/Objects/SidewaysMovingCat.cs

```diff
@@ -0,0 +1,45 @@
+using UnityEngine;
+using UnityEngine.Events;
+
+namespace Framework.Gameplay.MiniGames.Objects
+{
+    public sealed class SidewaysMovingCat : MonoBehaviour
+    {
+        [SerializeField , Range(0.5f , 2f)] private float movementStep = 0.5f;
+        [SerializeField, Tooltip("will be performed everytime the player decides to move")] private UnityEvent onMove = new();
+        [SerializeField] private UnityEvent onGoalReached = new();
+        [SerializeField] private Transform goalPosition;
```

Comment:
```suggestion
        [SerializeField , Range(0.5f , 2f)] private float movementStep = 0.5f;
        [SerializeField] private Transform goalPosition;
        [SerializeField, Tooltip("will be performed everytime the player decides to move")] private UnityEvent onMove = new();
        [SerializeField] private UnityEvent onGoalReached = new();
```

Zelf wil ik graag events onder aan de serialized fields. Ze kunnen veel inhoud hebben en dan mis je misschien de ene var die er onder staat.

+ [code conventie](https://github.com/bas-boop/ProjectIgnite/blob/main/CodeConvention.md#classes)